lib/controllers/cats.js

```diff
@@ -1,3 +1,20 @@
 const { Router } = require('express');
+const { cats } = require('../cats-data');
 
-module.exports = Router();
+module.exports = Router()
+  .get('/:id', (req, res) => {
+    let match;
+    for (const cat of cats) {
+      if (cat.id === req.params.id) {
+        match = cat;
+      }
+    }
+    return res.json(match);
+  })
+  .get('/', (req, res) => {
+    const filteredData = [];
+    for (const cat of cats) {
+      filteredData.push({ id: cat.id, name: cat.name });
+    }
```

Comment:
Same story here. For this section you could have used the `.map()` method.
```js
const filteredData = cats.map((cat) => ({ id: cat.id, name: cat.name }));

Reply:
Awesome thanks Vic!